File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Message/MessageAttachment.vb
Imports VecompSoftware.DocSuiteWeb.Data

<Serializable()>
Public Class MessageAttachment
    Inherits AuditableDomainObject(Of Int32)

#Region " Properties "

    Public Overridable Property Message As DSWMessage

    Public Overridable Property Archive As String

    Public Overridable Property ChainId As Integer

    Public Overridable Property DocumentEnum As Integer?

    Public Overridable Property Extension As String

#End Region

#Region " Constructors "

    Public Sub New()

    End Sub

    Public Sub New(archive As String, chainId As Integer, documentEnum As Integer?, extension As String)
        Me.Archive = archive
        Me.ChainId = chainId
        Me.DocumentEnum = documentEnum
        Me.Extension = extension
    End Sub

#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Messages/MessageAttachment.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.Messages
{
    public class MessageAttachment : DSWBaseEntity
    {
        #region [ Constructor ]

        public MessageAttachment() : this(Guid.NewGuid()) { }

        public MessageAttachment(Guid uniqueId)
            : base(uniqueId) { }
        #endregion

        #region [ Properties ]

        /// <summary>
        /// Archive in cui è archiviato l'attacment
        /// </summary>
        public string Archive { get; set; }
        /// <summary>
        /// Id della catena in cui è archiviato l'attacment
        /// </summary>
        public int ChainId { get; set; }
        /// <summary>
        /// Indice posizionale del documento
        /// </summary>
        public int? DocumentEnum { get; set; }
        /// <summary>
        /// Estensione dell'attachment
        /// </summary>
        public string Extension { get; set; }

        #endregion

        #region [ Navigation Properties ]
        /// <summary>
        /// Messaggio in cui è contenuto l'attachment
        /// </summary>
        public virtual Message Message { get; set; }
        #endregion
    }
}
